app/src/main/java/cz/muni/fi/pv256/movio2/fk410022/ui/adapter/MovieAdapter.java

```diff
@@ -0,0 +1,105 @@
+import cz.muni.fi.pv256.movio2.fk410022.model.store.FilmListType;
+import cz.muni.fi.pv256.movio2.fk410022.ui.drawable.GradientStar;
+import cz.muni.fi.pv256.movio2.fk410022.ui.listener.OnFilmClickListener;
+import cz.muni.fi.pv256.movio2.fk410022.util.Utils;
+import cz.muni.fi.pv256.movio2.fk410022.util.image.ImageHelper;
+
+public class MovieAdapter extends RecyclerView.Adapter<MovieAdapter.ViewHolder> {
+    private static final String TAG = MovieAdapter.class.getSimpleName();
+
+    private final OnFilmClickListener mListener;
+    private Film[] mDataset;
+    private FilmListType mType;
+
+    static class ViewHolder extends RecyclerView.ViewHolder {
+        TextView title;
+        ImageView imageView;
+        ImageView star;
+        RelativeLayout detail;
+        TextView rating;
+
+        private final Context context;
+
+        ViewHolder(View itemView) {
+            super(itemView);
+            context = itemView.getContext();
+
+            title = (TextView) itemView.findViewById(R.id.view_item_title);
+            imageView = (ImageView) itemView.findViewById(R.id.view_item_image);
+            star = (ImageView) itemView.findViewById(R.id.view_item_star);
+            detail = (RelativeLayout) itemView.findViewById(R.id.view_item_detail);
+            rating = (TextView) itemView.findViewById(R.id.view_item_rating);
+        }
+    }
+
+    public MovieAdapter(List<Film> myDataset, OnFilmClickListener listener, FilmListType type) {
+        this(myDataset.toArray(new Film[myDataset.size()]), listener, type);
+    }
+
+    public MovieAdapter(Film[] myDataset, OnFilmClickListener listener, FilmListType type) {
+        mDataset = myDataset;
+        mListener = listener;
+        mType = type;
+    }
+
+    @Override
+    public MovieAdapter.ViewHolder onCreateViewHolder(final ViewGroup parent,
+                                                      int viewType) {
+        View v = LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.view_item, parent, false);
+        return new ViewHolder(v);
+    }
+
+    @Override
+    public void onBindViewHolder(final ViewHolder holder, int position) {
+        final Context context = holder.context;
+        final Film film = mDataset[position];
+        holder.imageView.setImageDrawable(null); // clear
+        holder.detail.setBackgroundColor(Utils.addAlphaToColor(ContextCompat
+                .getColor(context, R.color.palette_default), 0.5f));
+
+        ImageHelper.displayPoster(film, new ImageViewAware(holder.imageView) {
+            @Override
+            protected void setImageBitmapInto(Bitmap bitmap, View view) {
+                super.setImageBitmapInto(bitmap, view);
+                Palette palette = Palette.from(bitmap).generate();
+                int muted = palette.getMutedColor(ContextCompat.getColor(context, R.color.palette_default));
+                holder.detail.setBackgroundColor(Utils.addAlphaToColor(muted, 0.5f));
```

Comment:
Nemyslím, resp. u mě to funguje (i v emulátoru). Podle mě to má co dělat s tou vyjímkou, že se ty obrázky ani nestáhnou.